beaconChain/typescript/helpers/stateTransition.ts

```diff
@@ -0,0 +1,195 @@
+// Helper functions related to state transition functions
+// TODO convert int (in return value for getActiveValidatorIndeces) to and int
+
+import {BeaconState, ShardAndCommittee, ValidatorRecord} from "../interfaces/state";
+import { ValidatorStatusCodes } from "../constants/enums";
+import constants from "../constants/constants"
+import {AttestationSignedData, BeaconBlock} from "../interfaces/blocks";
+
+type int = number;
+type bytes = number;
+type uint24 = number;
+type hash32 = number;
+
+/**
+ * The following is a function that gets active validator indices from the validator list.
+ * @param {ValidatorRecord[]} validators
+ * @returns {int[]}
+ */
+// TODO .map may cause mutation issues, probably want to look into fixing this later on...
+function getActiveValidatorIndices(validators: ValidatorRecord[]): int[] {
+  const result = validators.map((validator: ValidatorRecord, index: int) => {
+    if (validator.status === ValidatorStatusCodes.ACTIVE) {
+      return index;
+    }
+  });
+  return result;
+}
+
+/**
+ * The following is a function that shuffles any list; we primarily use it for the validator list.
+ * @param {any[]} values
+ * @param {hash32} seed
+ * @returns Returns the shuffled values with seed as entropy.
+ */
+// TODO finish this
+function shuffle(values: any[], seed: hash32): any[] {
+  const valuesCount: int = values.length;
+  // Entropy is consumed from the seed in 3-byte (24 bit) chunks.
+  const randBytes = 3;
+  // Highest possible result of the RNG
+  const randMax = 2 ** (randBytes * 8) - 1;
+
+  // The range of the RNG places an upper-bound on the size of the list that may be shuffled.
+  // It is a logic error to supply an oversized list.
+  if (valuesCount < randMax) throw new Error("Oversized list supplied to shuffle()!");
+
+  // Make a copy of the values
+  let output = values.slice();
+  const source = seed; // REALLY??
+  let index = 0; // REALLY??
+  while (index < valuesCount -1) {
+    // Re-hash the `source` to obtain a new pattern of bytes.
+    // TODO figure out what this hash function is in python -> JS
+    // let source = hash(source)
+
+    // Iterate through the `source` bytes in 3-byte chunks.
+
+  }
+
+  return []
+}
+
+/**
+ * Splits a list into split_count pieces.
+ * @param {any[]} values
+ * @param {int} splitCount
+ * @returns {any[]}
+ */
+export function split(values: any[], splitCount: int): any[] {
```

Comment:
I wonder if generics might worth while here, perhaps something like this:
```
export function split<T>(values: T[], splitCount: int): T[] {
  // Returns the split ``seq`` in ``split_count`` pieces in protocol.
  const listLength = values.length;
  let array: T[] = [];
  for (let i = 0; i < splitCount; i++) {
    array.push(values.slice(
      Math.floor((listLength * i) / splitCount), Math.floor((listLength * (i + 1)) / splitCount)
    ));
  }
  return array;
}
```

I do get an error on the above, but perhaps some food for thought

Reply:
+1 for generics. The use of `any` makes me squirm.